cou/apps/auxiliary.py

```diff
@@ -370,6 +375,41 @@ def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
             coro=set_require_osd_release_option(ceph_mon_unit.name, self.model),
         )
 
+    def _get_wait_until_noout_set_step(self) -> PreUpgradeStep:
```

Comment:
I have the feeling that this kind of check should happen when upgrading ceph-osd and not ceph-mon.

Ceph-mon is one of the first apps to be upgraded and is considered as control-plane. Don't we have to do this when we are going to upgrade ceph-osd to avoid re-balancing the data?

Reply:
This step is for the whole ceph cluster. But the operation is done on the control plane of the ceph cluster (i.e. ceph-mon). If we upgrade ceph-*, we should set noout